Gordon360/Services/AccountService.cs

```diff
@@ -170,6 +177,12 @@ public AccountViewModel GetAccountByUsername(string username)
 
             if (hall is not null) accounts = accounts.Where(a => a.Hall.StartsWithIgnoreCase(hall));
             if (classType is not null) accounts = accounts.Where(a => a.Class.StartsWithIgnoreCase(classType));
+            if (preferredClassYear is not null) accounts = accounts.Where(a => a.PreferredClassYear == (preferredClassYear));
+            if ((initialYear is not null) && (finalYear is not null))
+            {
+                accounts = accounts.Where(a => a.PreferredClassYear != "");
+                accounts = accounts.Where(a => Convert.ToInt32(a.PreferredClassYear) >= Convert.ToInt32(initialYear) && Convert.ToInt32(a.PreferredClassYear) <= Convert.ToInt32(finalYear));
```

Comment:
You could combine these so that you have to assign to `accounts` one less time
```suggestion
                accounts = accounts.Where(a => a.PreferredClassYear != "" && Convert.ToInt32(a.PreferredClassYear) >= Convert.ToInt32(initialYear) && Convert.ToInt32(a.PreferredClassYear) <= Convert.ToInt32(finalYear));
```

Reply:
I am not sure why exactly, but when we do the query in one line it has been giving us no results.

Reply:
About the view model instead of separate fields, we agree that it might be a lot of extra work for some validation that may not be necessary. Unless it is required or needed for us to change to the view model, I think we should leave it as is.